router/src/main/kotlin/io/moia/router/APIGatewayProxyEventExtensions.kt

```diff
@@ -56,9 +56,9 @@ fun APIGatewayProxyResponseEvent.withLocationHeader(request: APIGatewayProxyRequ
 
 fun APIGatewayProxyResponseEvent.location() = getHeaderCaseInsensitive("location")
 
-private fun getCaseInsensitive(key: String, map: Map<String, String>): String? =
-    map.entries
-        .firstOrNull { it.key.toLowerCase() == key.toLowerCase() }
+private fun getCaseInsensitive(key: String, map: Map<String, String>?): String? =
+    map?.entries
+        ?.firstOrNull { it.key.toLowerCase() == key.toLowerCase() }
```

Comment:
You could also use the equals function with the "ignore case" option as second parameter.
```suggestion
        ?.firstOrNull { key.equals(it.key, true) }
```